src/components/IspSelectWithIncidents/IspSelectWithIncidents.jsx

```diff
@@ -66,48 +72,127 @@ export default class IspSelectWithIncidents extends PureComponent {
   }
 
   /**
-   * convert array of ISPs to an array of options to display
+   * convert dictionary of ISPs to an array of options to display
    * @param {Array} isps ISPs to convert
    * @return {Array} array of {value: label:} objects
    */
-  getOptions(isps, selected) {
-    let options = isps;
-    if (selected && selected.length) {
-      options = isps.filter(isp => !selected.find(d => d.client_asn_number === isp.client_asn_number));
+  getOptions(isps, incidentData) {
+    let options = Object.values(isps);
+    options = options.map(isp => ({ value: isp.client_asn_number, label: isp.client_asn_name }));
+
+    if (incidentData) {
+      let i;
+      for (i of options) {
+        if (i.value in incidentData) {
+          i.hasInc = true;
+        }
+      }
+    }
+    return options;
+  }
+
+  showIncident(value) {
```

Comment:
Dockstring!

Reply:
Agree, adding more docstrings + comments would be great!